src/index.ts

```diff
@@ -34,28 +37,40 @@ namespace CommandIDs {
  * A service providing variable introspection.
  */
 const variableinspector: JupyterFrontEndPlugin<IVariableInspectorManager> = {
-  id: '@lckr/jupyterlab_variableinspector',
-  requires: [ICommandPalette, ILayoutRestorer, ILabShell],
+  id: '@lckr/jupyterlab_variableinspector:plugin',
+  requires: [ICommandPalette, ILayoutRestorer, ILabShell, ISettingRegistry],
   provides: IVariableInspectorManager,
   autoStart: true,
-  activate: (
+  activate: async (
     app: JupyterFrontEnd,
     palette: ICommandPalette,
     restorer: ILayoutRestorer,
-    labShell: ILabShell
-  ): IVariableInspectorManager => {
+    labShell: ILabShell,
+    settingRegistry: ISettingRegistry
+  ): Promise<IVariableInspectorManager> => {
     const manager = new VariableInspectorManager();
     const category = 'Variable Inspector';
     const command = CommandIDs.open;
     const label = 'Open Variable Inspector';
     const namespace = 'variableinspector';
     const tracker = new WidgetTracker<VariableInspectorPanel>({ namespace });
+    const settings = new PromiseDelegate<ISettingRegistry.ISettings>();
+
+    Promise.all([settingRegistry.load(variableinspector.id), app.restored])
+      .then(returnArray => {
+        return settings.resolve(returnArray[0]);
```

Comment:
To make the code more readable, you could use arg expansion:

```suggestion
      .then(([settings_]) => {
        return settings.resolve(settings_);
```